File: solution/075SortColors/075SortColors.go
package main

func sortColors(nums []int) {
	n := len(nums)
	if n <= 1 {
		return
	}
	i := 0
	j := n - 1
	k := 0
	for k <= j {
		if nums[k] < 1 {
			nums[i], nums[k] = nums[k], nums[i]
			i++
		} else if nums[k] > 1 {
			nums[j], nums[k] = nums[k], nums[j]
			j--
		} else {
			k++
		}
	}
}

func main() {

}

File: src/075SortColors/075SortColors.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let (mut i, mut j) = (0 as i32, nums.len() as i32 - 1);
        let mut k = 0;
        while k <= j {
            if nums[k as usize] < 1 {
                nums.swap(k as usize, i as usize);
                i += 1;
                k += 1;
            }else if nums[k as usize] > 1 {
                nums.swap(k as usize, j as usize);
                j -= 1;
            }else{
                k += 1;
            }
        }
    }
}
